app/src/main/java/com/example/sharingang/UserProfileViewModel.kt

```diff
@@ -135,6 +139,38 @@ class UserProfileViewModel @Inject constructor(
         }
     }
 
+    /**
+     * Resubscribe to all the topics of the user on login
+     * @param userId the id of the user
+     */
+    fun loginResubscribe(userId: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            val user = userRepository.get(userId)
+            if (user != null) {
+                val subs = ArrayList(user.subscriptions)
+                for (sub in subs) {
+                    if (sub != "") subscribeToTopic(sub)
+                }
```

Comment:
```suggestion
                ArrayList(user.subscriptions).forEach { if (it != "") subscribeToTopic(it)}
```
All this on one line (~ coverage)

Reply:
This code is tested right?
So it may decrease the coverage...

Reply:
This code isn't tested, it's the unsubscribe that is tested.